04/app.js

```diff
@@ -4,86 +4,87 @@ import { createRoot } from 'react-dom/client';
+		users: ['Jan Kowalski', 'Michał Nowak'],
+	};
 
-    renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
-                <li onClick={ this.clickHandler }>
-                    { name }
-                </li>
-            );
-        });
-    }
+	renderUsersList() {
+		const { users, searchQuery } = this.state;
+		let filtredUsers = [...users];
+		if (searchQuery) {
+			filtredUsers = users.filter((name) => name.includes(searchQuery));
+		}
+		return filtredUsers.map((name) => {
+			return <li onClick={this.clickHandler}>{name}</li>;
+		});
+	}
 
-    clickHandler = e => {
-        const {innerText: userName} = e.target;
-        this.removeUser(userName);
-    }
+	clickHandler = (e) => {
+		const { innerText: userName } = e.target;
+		this.removeUser(userName);
+	};
 
-    inputChange = e => {
-        const {name, value} = e.target;
-        this.setState({
-            [name]: value,
-        });
-    }
+	inputChange = (e) => {
+		const { name, value } = e.target;
+		this.setState({
+			[name]: value,
+		});
+	};
 
-    render() {
-        const { firstName, lastName } = this.state;
-        return (
-            <section onSubmit={ this.submitHandler }>
-                <form>
-                    <input name="firstName"
-                        value={ firstName }
-                        onChange={ this.inputChange }
-                    />
-                    <input name="lastName"
-                        value={ lastName }
-                        onChange={ this.inputChange }
-                    />
-                    <input type="submit"/>
-                </form>
-                <ul>{ this.renderUsersList() }</ul>
-            </section>
-        );
-    }
+	render() {
+		const { firstName, lastName, searchQuery } = this.state;
+		return (
+			<section onSubmit={this.submitHandler}>
+				<form>
+					<input
+						name="firstName"
+						value={firstName}
+						onChange={this.inputChange}
+					/>
+					<input name="lastName" value={lastName} onChange={this.inputChange} />
+					<input type="submit" />
+				</form>
+				<input
+					name="searchQuery"
+					value={searchQuery}
+					onChange={this.inputChange}
+				/>
+				<ul>{this.renderUsersList()}</ul>
+			</section>
+		);
+	}
 
-    submitHandler = e => {
-        e.preventDefault();
+	submitHandler = (e) => {
+		e.preventDefault();
 
-        const { firstName, lastName } = this.state;
-        if(firstName && lastName) {
-            this.addUser(`${firstName} ${lastName}`);
-            this.setState({
-                firstName: '',
-                lastName: '',
-            });
-        } else {
-            // tutaj komunikat dla użytkownika
-        }
-    }
+		const { firstName, lastName } = this.state;
+		if (firstName && lastName) {
+			this.addUser(`${firstName} ${lastName}`);
+			this.setState({
+				firstName: '',
+				lastName: '',
+			});
+		} else {
+			// tutaj komunikat dla użytkownika
```

Comment:
Może warto "obsłużyć" komunikat, aby użytkownik nie zastanawiał się dlaczego to nie działa ;)